src/gamma.jl

```diff
@@ -590,24 +590,45 @@ lgamma_r
     lfactorial(x)
 
 Compute the logarithmic factorial of a nonnegative integer `x`.
-Equivalent to [`lgamma`](@ref) of `x + 1`, but `lgamma` extends this function
+Equivalent to [`loggamma`](@ref) of `x + 1`, but `loggamma` extends this function
 to non-integer `x`.
 """
-lfactorial(x::Integer) = x < 0 ? throw(DomainError(x, "`x` must be non-negative.")) : lgamma(x + oneunit(x))
+lfactorial(x::Integer) = x < 0 ? throw(DomainError(x, "`x` must be non-negative.")) : loggamma(x + oneunit(x))
 Base.@deprecate lfact lfactorial
 
 """
-    lgamma(x)
+    logabsgamma(x)
 
-Compute the logarithm of the absolute value of [`gamma`](@ref) for
-[`Real`](@ref) `x`, while for [`Complex`](@ref) `x` compute the
-principal branch cut of the logarithm of `gamma(x)` (defined for negative `real(x)`
-by analytic continuation from positive `real(x)`).
+Compute the logarithm of absolute value of [`gamma`](@ref) for
+[`Real`](@ref) `x`and returns a tuple (logabsgamma(x), sign(gamma(x))).
 """
-function lgamma end
+function logabsgamma end
+logabsgamma(x::Real) = logabsgamma(float(x))
+
+"""
+    loggamma(x)
+
+Computes the logarithm of [`gamma`](@ref) for given `x`
+"""
+function loggamma end
+
+
+function loggamma(x::T) where {T<:Union{Float64,Float32}}
```

Comment:
this method can be removed since it is covered by `loggamma(x::Real)` below

Reply:
@simonbyrne I don't know why, but this throws `MethodError` saying no match for `loggamma(x:::Float64)` if I remove this.

Reply:
Can you paste the full error here?

Reply:
` Test threw exception
  Expression: loggamma(0.01) ≅ loggamma(0.01 + 0im) ≅ 4.599479878042022
  MethodError: no method matching loggamma(::Float64)
  Closest candidates are:
    loggamma(::AbstractFloat) at /home/sumegh/.julia/packages/SpecialFunctions/fvheQ/src/gamma.jl:803
    loggamma(::Real) at /home/sumegh/.julia/packages/SpecialFunctions/fvheQ/src/gamma.jl:625
    loggamma(::Number) at /home/sumegh/.julia/packages/SpecialFunctions/fvheQ/src/gamma.jl:629
    ...
  Stacktrace:
   [1] loggamma(::Float64) at /home/sumegh/.julia/packages/SpecialFunctions/fvheQ/src/gamma.jl:803
   [2] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1079
   [3] include at ./boot.jl:317 [inlined]
   [4] include_relative(::Module, ::String) at ./loading.jl:1044
   [5] include(::Module, ::String) at ./sysimg.jl:29
   [6] include(::String) at ./client.jl:392
   [7] top-level scope at none:0
   [8] eval(::Module, ::Any) at ./boot.jl:319
   [9] macro expansion at ./logging.jl:317 [inlined]
   [10] exec_options(::Base.JLOptions) at ./client.jl:219
   [11] _start() at ./client.jl:425
Other float types`